pkg/watermark/fetch/edge_fetcher_test.go

```diff
@@ -308,7 +308,7 @@ func TestBuffer_GetWatermarkWithMultiplePartition(t *testing.T) {
 			}
 			// this will always be 27 because the timeline has been populated ahead of time
```

Comment:
```suggestion
			// this will always be 26 because the timeline has been populated ahead of time
```

do you know why it became 26 from 27?

Reply:
because before for head watermark we were considering the max watermark across the partitions(27), now I updated it to consider only the incoming partition(26).